### PR TITLE
Update csv output for perf runs

### PR DESCRIPTION
**Pull request by lukewrites**
Change for internal consumption, making csv output from run friendly for perflab.

This PR adds the following CLI flags:
- `--perf-run`: takes no arguments; used for data sources that do not have `RunAsPerf` set in their `.ini` file.
- `--iteration`: takes a single integer argument; if unspecified, defaults to 0.

The `--perf-run` flag is complementary to existing logic that looks for a `RunAsPerf` boolean in a datasource `.ini` file. This means that the following commands will all work:

```terminal
$ python -m tdvt.tdvt run postgres_perf --iteration 9
$ python -m tdvt.tdvt run postgres_perf # iteration will default to 0
$ python -m tdvt.tdvt run postgres --perf-run --iteration 5
$ python -m tdvt.tdvt run postgres --perf-run  # iteration will default to 0
```

A TDVT run with `--perf-run` will change the output of `test_results_combined.csv`. The file will have different headers and will be pipe-delimited rather than comma-delimited.

This change means we no longer use `TabqueryCommandLineExtension` in order to do a perf run.

### CODE REVIEW

File: tdvt/tdvt/tdvt.py [hunk 1]
```diff
@@ -197,7 +196,7 @@ def run(self):
 
         start_time = time.time()
         self.test_config.thread_id = self.thread_id
-        failed_tests, skipped_tests, disabled_tests, total_tests = run_tests(self.test_config, self.test_set)
+        failed_tests, skipped_tests, disabled_tests, total_tests = run_tests(self.test_config, self.test_set,)
```
Comment (lukewrites) [inline tdvt/tdvt/tdvt.py:199]:
```suggestion
        failed_tests, skipped_tests, disabled_tests, total_tests = run_tests(self.test_config, self.test_set)
```

File: tdvt/tdvt/tdvt_core.py [hunk 2]
```diff
@@ -563,53 +603,105 @@ def get_csv_row_data(tds_name, test_name, test_path, test_result, test_case_inde
...
         columns.extend([actual_error, expected_error])
     return columns
 
-def write_csv_test_output(all_test_results, tds_file, skip_header, output_dir) -> Optional[Tuple[int, int, int, int]]:
+def write_csv_test_output(
+    all_test_results: Dict,
+    tds_file: str,
+    skip_header: bool,
+    output_dir: str,
+    perf_run: bool=False
```
Comment (rosswbrown) [inline tdvt/tdvt/tdvt_core.py:641]: Ideally this `perf_run` arg would be a formatter type object where we could (mostly) remove the if/else checks below and encapsulate the different output styles.

Roughly the same feedback for the is_perf_run and tdvt_dialect/perflab dialect

**Comment by lukewrites** [comment]
Closing this because it's so far out of date. I'll open a new work story for finally finishing and merging - got derailed when I went on leave.

**Comment by lukewrites** [comment]
Reopening, working on story now.

**Comment by lukewrites** [comment]
Because of the age of the story, I'm closing this PR in favor of #955

**Pull request closed without merging**